### ISSUE TITLE
Incorrect spacing with amsart

### ISSUE THREAD

**Issue by dpthurst** [henrikmidtiby/todonotes#36]
In some circumstances in amsart, the normal interline spacing is incorrectly handled with a \todo note. Specifically, this happens when a \todo note directly follows a subsection heading. (It may happen in other cases too, this is just one I was able to isolate.)

I've attached a MWE, in TeX format (t.txt) and the buggy output. I also attach the log file.

[t.pdf](https://github.com/henrikmidtiby/todonotes/files/2757449/t.pdf)
[t.txt](https://github.com/henrikmidtiby/todonotes/files/2757450/t.txt)
[t.log](https://github.com/henrikmidtiby/todonotes/files/2757459/t.log)

**Comment by dpthurst** [comment]
Here's another small example, using Definition environments.
[t-tex.txt](https://github.com/henrikmidtiby/todonotes/files/2757606/t-tex.txt)
[t.pdf](https://github.com/henrikmidtiby/todonotes/files/2757607/t.pdf)
[t.log](https://github.com/henrikmidtiby/todonotes/files/2757609/t.log)

**Comment by FrankMittelbach** [comment]
My guess is that you will have to live with this. The ams code around theorems has its own set of "weird" vertical spacing and it is basically really impossible for something like \todo to account for each and every variant out there. The way I see this command is that it does a reasonable job in most circumstances and usually manages to avoid changing the vertical spacing. In places where this is not really working, move it a line up or down (or inside the paragraph in your case). After all it is for your notes and those should typically be temp anyway.


Maybe Henrik wants to add a warning about the cases, but I doubt that they can (easily) be fixed.

**Comment by dpthurst** [comment]
Fair enough. In my case, I would have appreciated a warning: there was another bug that was my fault (involving using internals), and I would have been able to diagnose it faster with a warning about these cases.

**Comment by FrankMittelbach** [comment]
if you mean a warning in the package documentation then this is up to Henrik, but I agree it would help. 

If you mean when running the code then this is basically not possible (if it were then you could also fix it): basically the \todo adds something invisible there which upsets what later the theorem environment is doing ... and you can't easily determine that is is going to be the case from either end ... this is a messy area :-)

**Comment by FrankMittelbach** [comment]
see #37, the patch there fixes both issues
